### PR TITLE
[chore]Make get_field_attr safe and improve control flow in ts_core.py

### PR DESCRIPTION
**Pull request by dardevelin**
**Refactor get_field_attr in library.py and improve control flow in ts_core.py**

### Changes:

**1. Refactor `get_field_attr` in `library.py`:**
- Ensure safety using list slicing to avoid `IndexError`.
- Use `any` to determine if there is a value before using it.
- Use `.get()` for dictionary access to handle default cases and prevent `KeyError`.

**2. Improve control flow in `ts_core.py`:**
- Remove nesting by exiting early in `match_conditions`.
- Eliminate the need for try-catch blocks for better readability and control flow.

### Additional Fix:
- Included fix for non-merged pull request #318.

**Note:**
- Kept `existing_fields` case unchanged due to `add_field_to_entry`'s procedural nature.

### CODE REVIEW

**Comment by seakrueger** [comment]
FYI, in the future you don't need to close and open a new PR when you want to add new commits. You can just (sometimes, force) push to the same remote branch

**Comment by dardevelin** [comment]
@seakrueger I know, but I didn't know if id find what was the introduced issue on time, so to prevent people wasting time taking a look I closed it and created a new one when ready. cheers

**Review by eivl — APPROVED** [review]
Nice rewrite, Approved once my comments are resolved.

**Review by eivl — APPROVED** [review]
missclick.

File: tagstudio/src/core/library.py [hunk 1]
```diff
@@ -2167,14 +2167,15 @@ def mirror_entry_fields(self, entry_ids: list[int]) -> None:
...
-            return list(entry_field.keys())[0]
-        elif attribute.lower() == "content":
-            return entry_field[self.get_field_attr(entry_field, "id")]
-        else:
-            return self.get_field_obj(self.get_field_attr(entry_field, "id"))[
-                attribute.lower()
-            ]
+        match attribute.lower():
+            case "id":
+                return val[0] if not any(val := list(entry_field.keys())[:1]) else -1
```
[the remark this answers is not in the document]
Comment (eivl) [reply tagstudio/src/core/library.py:2172]: image

**Comment by yedpodtrzitko** [comment]
I'm afraid this PR is now obsolete when #332 has been merged.

**Pull request closed without merging**